File: java-automation/src/test/java/com/myecommerce/automation/dsl/protocols/CartProtocol.java
package com.myecommerce.automation.dsl.protocols;

import com.myecommerce.automation.dsl.domain.CartState;

public interface CartProtocol extends CatalogueProtocol {

    void viewCart();
    void addProductToCart();
    void removeFirstItemFromCart();
    void changeQuantityTo(int quantity);

    CartState getCartState();
}

File: kotlin-automation/src/test/kotlin/com/myecommerce/automation/dsl/protocols/CartProtocol.kt
package com.myecommerce.automation.dsl.protocols

import com.myecommerce.automation.dsl.domain.CartState

interface CartProtocol : CatalogueProtocol {
    fun viewCart()
    fun addProductToCart()
    fun removeFirstItemFromCart()
    fun changeQuantityTo(quantity: Int)

    fun getCartState(): CartState
}
